### ISSUE TITLE
Implement manual cropping with ffmpeg backend

### ISSUE THREAD

**Issue by balintlaczko** [fourMs/MGT-python#129]
The user experience should be similar to the existing method, so a little window should pop up showing the first frame, where the user can draw a cropping rectangle. Since cv2 is not so flexible with different codecs and formats, we can just export a temporary image with the first frame for this. After the selection is done, the *actual* cropping process should be ffmpeg-based, according to the cropping rectangle drawn by the user.

**Comment by balintlaczko** [comment]
Update: it seems to be possible to use ffmpeg's `cropdetect` for the automatic cropping, so it might be possible to implement the entire cropping module using ffmpeg (minus the popup window for drawing the crop rectangle for which cv2 is perfectly adequate).

**Comment by balintlaczko** [comment]
This was a little bit hard, and partly disappointing process, and the overall result with `cropdetect` is perhaps not as accurate as the earlier cv2-based solution (which is nevertheless still available in the repo). 

After *a lot* of testing, I landed on the following pipeline:
grayscale -> frame difference -> binary threshold -> cropdetect

the cv2-based version has this pipeline:
grayscale -> frame difference -> threshold (non-binary) -> median filter -> "cropdetect"

Although the cv2-based version is a bit more accurate, I anyway decided to go for the ffmpeg-based solution since it is *much* faster. To compensate for `cropdetect`'s tendency to crop too much I raised the `motion_box_margin`, so that we might not crop so "tightly", but we at least don't crop away important stuff. Another benefit of having it ffmpeg-based is that with this all preprocessing steps have become ffmpeg-based, so there is no need (at least at this stage) to convert everything to avi, which can be a huge timesaver when it comes to long videos.

Bottom line: I implemented the manual cropping mode with ffmpeg, with only showing the "popup" cropping window using cv2, and the automatic cropping with using entirely ffmpeg.
